components/sync/service/brave_sync_service_impl.h

```diff
@@ -124,6 +124,11 @@ class BraveSyncServiceImpl : public SyncServiceImpl {
 
   int completed_cycles_count_ = 0;
 
+  // This flag is set to true during BraveSyncServiceImpl::Initialize call. The
+  // reason is that upstream SyncServiceImpl::Initialize() can invoke
+  // StopAndClear, but we don't want to invoke AddLeaveChainDetail in that case
+  bool executing_initialize_ = false;
```

Comment:
can we rename this to be `is_initializing`?

Reply:
yes, done https://github.com/brave/brave-core/pull/22456/commits/7a281051cfa9d05435ad9b3390161f9e003548e3

thanks, this sounds better